File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_java_labs_and_solutions/factory_method/solution/OrganicTomato.java
// OrganicTomato.java

public class OrganicTomato extends Tomato
{
	public OrganicTomato(double aPrice, int aNumberOfInsects)
	{
		super(aPrice);
		myNumberOfInsects = aNumberOfInsects;
	}

	public OrganicTomato(double aPrice)
	{
		super(aPrice);
		myNumberOfInsects = 0;
	}

	private int myNumberOfInsects;
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_cpp_labs_and_solutions/factory_method/solution/organic_tomato.cpp
// organic_tomato.cpp

#include "organic_tomato.h"
#include "tomato.h"

organic_tomato::organic_tomato(double a_price, int a_number_of_insects)
: tomato(a_price)
{
	my_number_of_insects = a_number_of_insects;
}

organic_tomato::~organic_tomato()
{
}
